api/app/controllers/spree/api/address_book_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Api
+    class AddressBookController < Spree::Api::BaseController
+      # Note: the AddressBook is the resource to think about here, not individual addresses
+
+      before_filter :ensure_api_token
+
+      def mine
+         @user_addresses = current_api_user.user_addresses
+      end
+
+      def update
+        address_params = address_book_params
+        default_flag = address_params.delete(:default)
+        address = current_api_user.save_in_address_book(address_params, default_flag)
+        if address.valid?
+          @user_addresses = current_api_user.user_addresses
+          render :mine, status: :ok
```

Comment:
As I mentioned in first comment - this is good time to discuss adding 303 to project, because you added new code, where we can use it.